Octree/include/CGAL/Octree/Node.h

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2007-2020  INRIA (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Jackson Campolattaro, Cédric Portaneri, Tong Zhao
+
+#ifndef CGAL_OCTREE_NODE_H
+#define CGAL_OCTREE_NODE_H
+
+#include <CGAL/license/Octree.h>
+
+#include <boost/range/iterator_range.hpp>
+
+#include <array>
+#include <memory>
+#include <bitset>
+#include <cassert>
+
+namespace CGAL {
+namespace Octree {
+namespace Node {
```

Comment:
Do we really want namespaces for Node and Walker?   `namespace Node` only contains the `class Node`.

Reply:
I'm a bit uncertain about this too, I agree that the redundant namespaces are a bit ugly.

I originally added `namespace Node` in order to be able to typedef it inside the Octree without introducing a naming conflict:
https://github.com/JacksonCampolattaro/cgal/blob/d290ddcfe792db0b26a462de592054181666134b/Octree/include/CGAL/Octree.h#L96
Is there a better way to avoid that kind of verbosity?

I added the Walker namespace when I noticed that I had several functors with the suffix "walker" on their name. E.g. `Preorder_walker` became `Walker::Preorder`. I think a namespace is appropriate here because it makes it clear that all the structs contained are different versions of the same concept.

Reply:
I agree with Andreas that the namespace Node is not a good idea. For the typedef, I think you can trick it by doing `typedef CGAL::Octree::Node<blabla> Node;` but I'm not sure. Otherwise, just use another name (I agree that it's less satisfying but it's better than adding a namespace just for the sake of typedefing later).

For Walker, as we already discussed, I agree with Jackson, this namespace have / will have several classes and serves for a clear naming `Walker::Preorder`, `Walker::Postorder`, etc.

Reply:
In the AABB Tree package, we have a `Traversal_traits`.   So you might replace `Walker` with `Traversal`.

Reply:
I just removed the `Node` namespace, @sgiraudot Thanks for the suggestion, the trick worked!